did-tezos/src/lib.rs

```diff
@@ -130,16 +201,162 @@ impl DIDMethod for DIDTz {
     }
 }
 
+#[derive(Deserialize)]
+#[serde(rename_all = "kebab-case")]
+struct SignedIetfJsonPatchPayload {
+    ietf_json_patch: serde_json::Value,
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all = "kebab-case")]
+#[serde(tag = "type", content = "value")]
+enum Updates {
+    SignedIetfJsonPatch(Vec<String>),
+}
+
+impl DIDTz {
+    fn tier1_derivation(
+        did: &str,
+        vm_didurl: &DIDURL,
+        proof_type: &str,
+        address: &str,
+        network: &str,
+    ) -> Document {
+        Document {
+            context: Contexts::One(Context::URI(DEFAULT_CONTEXT.to_string())),
+            id: did.to_string(),
+            authentication: Some(vec![VerificationMethod::DIDURL(vm_didurl.clone())]),
+            assertion_method: Some(vec![VerificationMethod::DIDURL(vm_didurl.clone())]),
+            verification_method: Some(vec![VerificationMethod::Map(VerificationMethodMap {
+                id: String::from(vm_didurl.clone()),
+                type_: proof_type.to_string(),
+                controller: did.to_string(),
+                blockchain_account_id: Some(format!("{}@tezos:{}", address.to_string(), network)),
+                ..Default::default()
+            })]),
+            ..Default::default()
+        }
+    }
+
+    async fn tier2_resolution(did: &str, address: &str, network: &str) -> Result<Service> {
+        let did_manager = explorer::retrieve_did_manager(address, network).await?;
+        Ok(explorer::execute_service_view(did, &did_manager, network).await?)
+    }
+
+    async fn tier3_updates(
+        &self,
+        prefix: &str,
+        doc: &mut Document,
+        updates: Updates,
+        public_key: Option<String>,
```

Comment:
Do we assume that all updates are signed by the same public key? Instead, we should expect that updates are signed by the previous public key as part of DID resolution up to that point. In your loop, we should therefore be updating the public key to the latest one found in the document. Am I missing something?

Reply:
We had conversations about offline updates and the impossibility to derive the public key. I'll fetch the public key from the DID Manager and use it instead of requiring the public key to be passed through the resolution input metadata.

Reply:
Wait no we said `verification_method` in the DID Manager is an address -- I get confused with the different verification methods.

Reply:
The signed ietf patch updating a verification method should potentially affect the expected key for the next patch was my point

Reply:
Done in 77e0011 but it's bloody ugly, will need to be extensively cleaned up and tested for chained updates.

Reply:
Seems like a reasonable approach to me, thanks! I think in the future, if we consider a JSON-LD Proof approach to signed IETF JSON patches, then we could possibly remove the requirement of public keys in tz2 and tz3. I have started an issue to track this.